azurecontainerapps.ts

```diff
@@ -621,11 +618,9 @@ export class azurecontainerapps {
         // Ensure '-i' argument and '--source' argument are not both provided
         if (!this.util.isNullOrEmpty(this.imageToDeploy)) {
             this.commandLineArgs.push(`-i ${this.imageToDeploy}`);
-        } else if (this.shouldCreateOrUpdateContainerAppWithUp) {
+        } else if (!this.util.isNullOrEmpty(this.appSourcePath) && this.useInternalRegistry) {
             this.commandLineArgs.push(`--source ${this.appSourcePath}`);
-            this.commandLineArgs.push(`-l ${this.location}`);
```

Comment:
is location arg not allowed / not working for `container app update`?

Reply:
nope `update` does not support location. It derives it from env